libraries/AP_HAL_Linux/OpticalFlow_Onboard.cpp

```diff
@@ -71,18 +84,10 @@ void OpticalFlow_Onboard::init(AP_HAL::OpticalFlow::Gyro_Cb get_gyro)
                                    V4L2_MBUS_FMT_UYVY8_2X8)) {
         AP_HAL::panic("OpticalFlow_Onboard: couldn't set subdev fmt\n");
     }
-    const char* device_path = HAL_OPTFLOW_ONBOARD_VDEV_PATH;
-    memtype = V4L2_MEMORY_MMAP;
-    nbufs = HAL_OPTFLOW_ONBOARD_NBUFS;
-    _width = HAL_OPTFLOW_ONBOARD_OUTPUT_WIDTH;
-    _height = HAL_OPTFLOW_ONBOARD_OUTPUT_HEIGHT;
-    crop_width = HAL_OPTFLOW_ONBOARD_CROP_WIDTH;
-    crop_height = HAL_OPTFLOW_ONBOARD_CROP_HEIGHT;
-    top = 0;
-    /* make the image square by cropping to YxY, removing the lateral edges */
-    left = (HAL_OPTFLOW_ONBOARD_SENSOR_WIDTH -
-            HAL_OPTFLOW_ONBOARD_SENSOR_HEIGHT) / 2;
     _format = V4L2_PIX_FMT_NV12;
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_MINLURE
+    _format = V4L2_PIX_FMT_YUYV;
+#endif
```

Comment:
In order for this to be more generic, couldn't we query the supported formats from the camera we are using and set to one of the formats that we support?  You picked a random usb camera, other people may have other cameras with different formats.